src/components/Cluster/ClusterDetail/ClusterDetail.js

```diff
@@ -11,37 +11,47 @@ import { AppRoutes } from 'shared/constants/routes';
 import GettingStarted from '../../GettingStarted/GettingStarted';
 import ClusterDetailView from './ClusterDetailView';
 
-const ClusterDetail = (props) => {
+const ClusterDetail = ({ match, cluster, clusterId, ...rest }) => {
   const dispatch = useDispatch();
 
-  if (!props.cluster) {
-    new FlashMessage(
-      `Cluster <code>${props.clusterId}</code> doesn't exist.`,
-      messageType.INFO,
-      messageTTL.MEDIUM
-    );
+  useEffect(() => {
```

Comment:
Using an effect because executing updates during rendering (such as the `dispatch(push(AppRoutes.Home));` we had before) is forbidden by React, and will throw an error.